src/core/src/runtime/tensor.cpp

```diff
@@ -162,26 +162,44 @@ ov::Shape calc_static_shape_for_file(const std::filesystem::path& file_name,
     dynamic_dimension = Dimension(new_dimension_size);
     return partial_shape_copy.get_shape();
 }
+
+void read_tensor_data(const std::filesystem::path& file_name, Tensor& tensor, size_t offset) {
+    OPENVINO_ASSERT(tensor.get_element_type() != ov::element::string);
+    std::ifstream fin(file_name, std::ios::binary);
+    fin.seekg(offset);
+    auto bytes_to_read = tensor.get_byte_size();
+    fin.read(static_cast<char*>(tensor.data()), bytes_to_read);
+    OPENVINO_ASSERT(static_cast<size_t>(fin.gcount()) == bytes_to_read,
+                    "Cannot read ",
+                    bytes_to_read,
+                    "bytes from ",
+                    file_name);
+}
 }  // namespace
 
 Tensor read_tensor_data(const std::filesystem::path& file_name,
                         const ov::element::Type& element_type,
                         const ov::PartialShape& partial_shape,
-                        size_t offset_in_bytes) {
+                        size_t offset_in_bytes,
+                        bool mmap) {
```

Comment:
Can we have separate function to read using mmap?

e.g when user want to read using mmap usage is straight forward just `read_tensor_data(path)`
but when mmap is optione then is required:


`read_tensor_data(path, type, shape, offset, mmap)` 

instead just do:

`mmap ? read_mmap_tensor_data(path) : read_tensor_data(path)`